### ISSUE TITLE
Request for example on how to use IAM Auth tokens for RDS access

### ISSUE THREAD

**Issue by c-p-i-o** [awslabs/aws-sdk-rust#792]
### Describe the issue

This is a request for adding an example on how to correctly use [generate-db-auth-token](https://docs.aws.amazon.com/cli/latest/reference/rds/generate-db-auth-token.html) in Rust for DB operations.
I followed the example as listed [here](https://github.com/awslabs/aws-sdk-rust/issues/147) and am able to generate IAM tokens. However, under the covers I'm using `sqlx` to make connections to a DB. Access works correctly when I use `password`, however I'm currently unable to get it to work when I use the `IAM` token.
I suspect I probably need to follow what was done in [golang](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/rdsutils/#BuildAuthToken)? 

### Links

https://docs.aws.amazon.com/sdk-for-rust/latest/dg/rust_sts_code_examples.html

**Comment by Velfi** [comment]
@c-p-i-o We don't have a crate for this and we don't plan on releasing something soon. Your idea to follow what the golang SDK does is a good one. I'm guessing that this is just a matter of formatting a string correctly.

@DavidSouther Perhaps this use-case is worthy of an example?

**Comment by rcoh** [comment]
I agree an end-to-end worked example would be good—in the meantime, this should get you most of the way there: https://docs.rs/aws-sig-auth/0.55.1/aws_sig_auth/#generate-rds-iam-token

**Comment by lcmgh** [comment]
Stumbled over this issue as https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.Connecting.html was not helpful for Rust.

**Comment by lcmgh** [comment]
> https://docs.rs/aws-sig-auth/0.55.1/aws_sig_auth/#generate-rds-iam-token


How to go from a `AssumeRoleProvider` to `&Credentials` ?

**Comment by lcmgh** [comment]
> 

Don't you have to use the IAM role name as username and the generated token as password?

**Comment by sammoh94** [comment]
@lcmgh The issue seems to be that the token that is being returned by the aws_sign_auth crate doesn't contain a Security-Token which I imagine is the essential component to being able to use it as the password

**Comment by sammoh94** [comment]
@lcmgh Would you mind posting what you passed in to assume_role_provider for the proxy argument? Thanks!

**Comment by lcmgh** [comment]
> @lcmgh Would you mind posting what you passed in to assume_role_provider for the proxy argument? Thanks!

```rust
/// Returns `ProxyConnector<HttpConnector>` if env. variable 'https_proxy' is set
pub fn determine_proxy() -> Option<ProxyConnector<HttpConnector>> {
    let proxy_url: Url = env::var("https_proxy").ok()?.parse().ok()?;
    let proxy_uri: Uri = env::var("https_proxy").ok()?.parse().ok()?;
    let mut proxy = Proxy::new(Intercept::All, proxy_uri);

    if let Some(password) = proxy_url.password() {
        proxy.set_authorization(Authorization::basic(proxy_url.username(), password));
    }

    let connector = HttpConnector::new();
    Some(ProxyConnector::from_proxy(connector, proxy).unwrap())
}
```

**Comment by ysaito1001** [comment]
A utility available in `aws_sdk_rds` crate https://docs.rs/aws-sdk-rds/latest/aws_sdk_rds/auth_token/struct.AuthTokenGenerator.html
